### PR TITLE
Unify total and female TN aggregate reports and add a site scraper

### PR DESCRIPTION
**Pull request by ohinds**

### CODE REVIEW

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 1]
```diff
@@ -38,7 +39,12 @@
 
 
 def parse(filename: str) -> Dict[DeclarativeMeta, pd.DataFrame]:
-    table = _parse_table(filename)
+    # There are two types of reports, total jail population and female
```
Comment (jessex) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Does the total jail population report include male + female?

Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Yup! We can handle that in the view generation.

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 2]
```diff
@@ -68,47 +76,70 @@ def _parse_table(filename: str) -> pd.DataFrame:
...
-        r'Pre- trial Felony': 'pretrial_felony_population',
-        r'Pre- trial Misd\.': 'pretrial_misdemeanor_population',
-        r'Total Jail Pop\.': 'total_jail_population',
-        r'Total Beds\*\*': 'total_beds',
-    }
+    if is_female:
+        rename = {
+            r'FACILITY': 'facility_name',
+            r'TDOC Backup.*': 'tdoc_backup_population',
+            r'FEMALE POPULATION': 'smashed',
```
Comment (zacharylawrence) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Sharing the word `smashed` between these 2 dfs is a little magical. Do you think it makes sense to either pull the logic out into a separate step that more explicitly handles both the `is_female` and `not is_female` cases? At the least, I think we should pull 'smashed' out into a variable to make it more explicit what's happening here.

File: recidiviz/persistence/database/schema.py [hunk 3]
```diff
@@ -963,3 +963,24 @@ class TnFacilityAggregate(Base, _AggregateTableMixin):
     pretrial_misdemeanor_population = Column(Integer)
     total_jail_population = Column(Integer)
     total_beds = Column(Integer)
+
+
+class TnFacilityFemaleAggregate(Base, _AggregateTableMixin):
```
Comment (zacharylawrence) [inline recidiviz/persistence/database/schema.py]: Could you also add the migration to this PR and run it after merging? As per @zdg2102's instructions.

Comment (ohinds) [reply recidiviz/persistence/database/schema.py]: Done!

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 2 above)
Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Done!

File: recidiviz/persistence/database/schema.py (hunk 3 above)
Comment (zacharylawrence) [reply recidiviz/persistence/database/schema.py]: LGTM! @ohinds Don't forget to run this at the same time that you merge this into master!

**Pull request merged**